### PR TITLE
Add README.md for `retry` package.

### PR DESCRIPTION
**Pull request by ogaca-dd**
### What does this PR do?

Add README.md for `retry` package.

### Additional Notes

The branch is based from https://github.com/DataDog/datadog-agent/pull/7703.

### CODE REVIEW

File: pkg/forwarder/internal/retry/README.md
Comment (sgnn7) [inline pkg/forwarder/internal/retry/README.md:31]:
```suggestion
When those transactions are sent, `File 2` content is read and removed, sending the transactions `tr4` and `tr3`.
```

File: pkg/forwarder/internal/retry/README.md
Comment (sgnn7) [inline pkg/forwarder/internal/retry/README.md:32]:
```suggestion
Finally, `File 1` is read and removed, sending the `tr2` and `tr1` to the intake.
```

File: pkg/forwarder/internal/retry/README.md
Comment (sgnn7) [inline pkg/forwarder/internal/retry/README.md:34]:
```suggestion
![Removing transactions from the retry queue](images/Extract.png)
```

File: pkg/forwarder/internal/retry/README.md
Comment (sgnn7) [inline pkg/forwarder/internal/retry/README.md:40]:
```suggestion
* At agent startup, previous files are reloaded. Unknown domains and old files are removed.
```

**Review by sgnn7 — CHANGES_REQUESTED** [review]
@ogaca-dd Lots of non-blocking nit suggestions but overall great work on documenting this! 

I did notice that the images don't seem to show up [when reading it](https://github.com/DataDog/datadog-agent/tree/olivierg/add-readme-for-retry/pkg/forwarder/internal/retry) so that's the only thing blocking this PR for me. It also may be good to have this doc linked from the [main forwarder README](https://github.com/DataDog/datadog-agent/blob/olivierg/add-readme-for-retry/pkg/forwarder/README.md).

**Comment by ogaca-dd** [comment]
> I did notice that the images don't seem to show up when reading it so that's the only thing blocking this PR for me. 

The link is resolved as https://github.com/DataDog/datadog-agent/blob/master/pkg/forwarder/internal/retry/images/Enqueue.png. As the PR is not yet merged, the image is not displayed.

I used the same method as for https://github.com/DataDog/datadog-agent/blob/bb61ba44b0199184f02bd8b8054b466a1dc7ac39/docs/dogstatsd/internals.md#L12.

Once the PR is merged, the image should be displayed.
Let me know if you know a better technique that still use a relative path.

File: releasenotes/notes/fixwinprocesshandleshang-bce162f5c0fee73f.yaml
Comment (kayayarai) [inline releasenotes/notes/fixwinprocesshandleshang-bce162f5c0fee73f.yaml:11]:
```suggestion
    On Windows, fixes problem in process agent wherein Windows processes
```

**Review by kayayarai — APPROVED** [review]
One tiny capitalization edit

File: releasenotes/notes/fixwinprocesshandleshang-bce162f5c0fee73f.yaml
Comment (ogaca-dd) [reply releasenotes/notes/fixwinprocesshandleshang-bce162f5c0fee73f.yaml]: I did a wrong merge and this change is not part of my development.

**Comment by ogaca-dd** [comment]
Closed in favor of https://github.com/DataDog/datadog-agent/pull/7848

**Pull request closed without merging**
